server/controllers/profile.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Controller.
+ */
+module.exports = function(app) {
+
+  const models = app.set('models');
+  const errors = app.errors;
+
+  return (req, res, next) => {
+    const slug = req.params.slug;
+    const controllers = req.app.set('controllers');
+
+    models.Group
+      .findOne({where: { slug }})
+      .then((group) => {
+        if(group) {
+          req.group = group;
+          return controllers.groups.getOne(req, res, next);
+        }
+        else {
+          models.User.findOne({where: {username: slug}})
+          .then((user) => {
+            if(user) {
+            req.user = user;
+            req.query.profile = true;
+            return controllers.users.show(req, res, next);
```

Comment:
@asood123 The complexity comes from mixing users and groups within the same endpoint, although they are resources of different nature. I find this query counterintuitive: `User.findOne({where: {username: slug}})`. A slug isn't a username. What if a user has the same username as a group's slug? Then it will be shadowed.

Haven't been in the loop for profiles but AFAIK using separate routes for group and user profiles would provide better separation of concerns. Then code complexity above would collapse.